samples/zigbee/ncp/sample.yaml

```diff
@@ -13,16 +13,44 @@ tests:
       - nrf21540dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
 
-  zigbee.ncp.usb:
+  zigbee.ncp.usb.nrf52840dk_nrf52840:
     build_only: true
-    platform_allow: nrf52833dk_nrf52833 nrf52840dk_nrf52840 nrf52840dongle_nrf52840 nrf21540dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840
     tags: ci_build ncp
-    extra_args: CONF_FILE=prj_usb.conf
+    extra_args: CONF_FILE=prj_usb.conf DTC_OVERLAY_FILE=boards/nrf52840dk_nrf52840_usb.overlay
     integration_platforms:
       - nrf52840dk_nrf52840
+
+  zigbee.ncp.usb.nrf52833dk_nrf52833:
+    build_only: true
+    platform_allow: nrf52833dk_nrf52833
+    tags: ci_build ncp
+    extra_args: CONF_FILE=prj_usb.conf DTC_OVERLAY_FILE=boards/nrf52833dk_nrf52833_usb.overlay
```

Comment:
I remember that there was some magic, pulling in overlays, based on the board name and the main overlay suffix (`usb` in this case). Does it cover the DTC overlays or it does not, so we need to manually iterate through those variants?

Reply:
The reason for the new entries is that we can no longer configure all board "chosen" devices using `prj_usb.conf`, it happens in DT now. The overlay files following the board name are used by the non-USB case. For the USB case we need to manually specify a different overlay file for each board (since not all USB overlays have the same content), hence the need for individual test cases.

Reply:
There are discussions upstream to introduce a new "snippets" feature to precisely improve this. cc: @jfischer-no 

Is this `build_type` a supported feature? I've never heard about it

Reply:
> Is this `build_type` a supported feature? I've never heard about it
It isn't for DTC overlay.
 
If I am not mistaken, this is what is supported (following https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_multi_image.html?highlight=child%20image#image-specific-variables):
```
|-- boards DIRECTORY
|   |-- <board>.conf
|   |-- <board>_<buildtype>.conf
|   |-- <board>.overlay
|   |-- <board>_<revision>.overlay
```

What IMO should also be supported:
```
<board>_<buildtype>.overlay
```

This would make it at least consistent with Kconfig overlay files. If they are supported, so should be DTC.

Reply:
> There are discussions upstream to introduce a new "snippets" feature to precisely improve this. cc: @jfischer-no
>
> Is this build_type a supported feature? I've never heard about it


it's a supported feature and documented here.
https://docs.zephyrproject.org/latest/build/kconfig/setting.html#initial-conf
> Otherwise if CONF_FILE is set, and a single configuration file of the form prj_\<build\>.conf is used, then if file boards/\<BOARD\>\_\<build\>.conf exists in same folder as file prj_\<build\>.conf, the result of merging prj_\<build\>.conf and boards/\<BOARD\>_\<build\>.conf is used.

It was initially introduced as a need for testcase variation, but as interest grows I have had discussions with @SebastianBoe on refactoring this and make it more visible with dedicated CMake argument, just not had the time yet.

Reply:
> What IMO should also be supported:
> 
> ```
> <board>_<buildtype>.overlay
> ```
> 
> This would make it at least consistent with Kconfig overlay files. If they are supported, so should be DTC.

@sebastiandraus there was a PR proposing this, but as no one seemed to have interest in it, it didn't make it far.
Feel free to make a comment in the closed PR, and it can be revived.

Reply:
so can we leave this for a future improvement if the feature gets in?

Reply:
@sebastiandraus since you are planning to make a follow-up PR, would you play around this idea there too?

Reply:
> @gmarull Can we revert these changes in sample.yaml? They cause some of our Jenkins plans to fail. Instead of having variants like `zigbee.ncp.usb.nrf52833dk_nrf52833:` we can have single `sample.zigbee.ncp.usb:` like we used to have. I would suggest that setting the board specific devicetree overlay for `.usb` variant is set in the sample's CMakeLists, similarly to this: [#8084 (files)](https://github.com/nrfconnect/sdk-nrf/pull/8084/files#diff-30f966b840fa11767af299a8201b67e8b24ad3a32489eafb082338c71b03fb98) This is would be the logic that allows to automagically include buildtype specific devicetree overlays, like so: `<board>_<buildtype>.overlay` Once it is allowed globally we can simply remove local logic from the sample's CMakeLists

To be honest, this CMake thing looks like an ad-hoc solution that won't scale and should not be introduced in all samples, but rather find a proper solution. If there are failures now, they should just be fixed (note that CI is green), how can I reproduce them?

Reply:
cc: @tejlmand opinions on the CMake approach?